orgs/organization.go

```diff
@@ -0,0 +1,144 @@
+package orgs
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/LF-Engineering/dev-analytics-libraries/auth0"
+	"github.com/LF-Engineering/dev-analytics-libraries/elastic"
+	"github.com/LF-Engineering/dev-analytics-libraries/http"
+)
+
+// Org struct
+type Org struct {
+	OrgBaseURL       string
+	ESCacheURL       string
+	ESCacheUsername  string
+	ESCachePassword  string
+	AuthGrantType    string
+	AuthClientID     string
+	AuthClientSecret string
+	AuthAudience     string
+	AuthURL          string
+	Environment      string
+	httpClient       *http.ClientProvider
+	esClient         *elastic.ClientProvider
+	auth0Client      *auth0.ClientProvider
+}
+
+// SearchOrganization ...
+func (o *Org) SearchOrganization(name string, pageSize string, offset string) *SearchOrganizationResponse {
+	if name == "" {
+		log.Println("SearchOrganization: name is empty")
+		return nil
+	}
+	token, err := o.auth0Client.ValidateToken(o.Environment)
+	if err != nil {
+		log.Println(err)
+	}
+	headers := make(map[string]string, 0)
+	headers["Content-type"] = "application/json"
+	headers["Authorization"] = fmt.Sprintf("%s %s", "Bearer", token)
+
+	if offset == "" {
+		offset = "0"
+	}
+	if pageSize == "" {
+		pageSize = "100"
+	}
+	endpoint := o.OrgBaseURL + "/orgs/search?name=" + name + "&pageSize=" + pageSize + "&offset=" + offset
+	_, res, err := o.httpClient.Request(strings.TrimSpace(endpoint), "GET", headers, nil, nil)
+	if err != nil {
+		log.Println("SearchOrganization: Could not get the organization list: ", err)
+		return nil
+	}
+	var response SearchOrganizationResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		log.Println("SearchOrganization: failed to unmarshal SearchOrganizationResponse: ", err)
+		return nil
+	}
+	return &response
+}
+
+// LookupOrganization ...
+func (o *Org) LookupOrganization(name string) *Organization {
+	if name == "" {
+		log.Println("LookupOrganization: name is empty")
+		return nil
+	}
+	token, err := o.auth0Client.ValidateToken(o.Environment)
+	if err != nil {
+		log.Println(err)
```

Comment:
return error to be handled in consumer side

Reply:
done!